larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc

```diff
@@ -0,0 +1,472 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc
+ *
+ *  @brief  Implementation file for the reclustering algorithm class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "Managers/ClusterManager.h"
+
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+#include "larpandoracontent/LArHelpers/LArPcaHelper.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+ThreeDReclusteringAlgorithm::ThreeDReclusteringAlgorithm():
+    m_PfosForReclusteringListName("newShowerParticles3D"),
+    m_hitThresholdForNewPfo(0),
+    m_fomThresholdForReclustering(0.3)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+ThreeDReclusteringAlgorithm::~ThreeDReclusteringAlgorithm()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode ThreeDReclusteringAlgorithm::Run()
+{
+    //Only proceed if successfuly get shower pfo list, and it has at least one pfo
+    const PfoList *pShowerPfoList(nullptr);
+    if (!((STATUS_CODE_SUCCESS == PandoraContentApi::GetList(*this, "ShowerParticles3D", pShowerPfoList)) && pShowerPfoList->size()))
+    {
+        return STATUS_CODE_SUCCESS;
+    }
+
+    m_PfosForReclusteringListName = "newShowerParticles3D";
```

Comment:
If I understand correctly, there is a way to set this via xml, so this line overwrites any other setting? Does it need to be removed? It is also set in the class constructor so it has a defined value if no xml call is made.

Reply:
Thanks for spotting this!

Reply:
Actually, realised there was some confusion here. The list name I want to specify via XML is "ShowerParticles3D". m_PfosForReclusteringListName is an internal one. I have modified the algorithm to reflect this. Thanks again for noticing this!